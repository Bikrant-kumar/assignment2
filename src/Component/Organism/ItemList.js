import React from 'react';
import ItemCard from '../Molecules/ItemCard';
import Grid from '@material-ui/core/Grid';
import Box from '@material-ui/core/Box';
import Label from '../Atoms/Label';


function ItemList(props) {
    
    return (
        <Box >
            <Label text="MENU"/>
            <Grid container
            item xs={12}>
            {
                props.items && 
                props.items.map((item) => <ItemCard item={item} size="small" color="primary" text="Delete"  manager={props.manager}/>)
            }
            </Grid>
        </Box>
    )
}

export default ItemList