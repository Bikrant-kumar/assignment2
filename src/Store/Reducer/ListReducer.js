export const initState = {    
   
};

export function itemReducer (state = initState, action) {
    switch (action.type) {
        case "DELETE_ITEM":
            const newItem = state.item.filter((item) => item.id !== action.id);
            return{
                ...state,
                item: newItem
            };

        case "DELETE_ROW":
            const newTable= action.table.item.splice(action.table.item.indexOf(action.rowId),1)
                action.table.totalCost = action.table.totalCost - action.rowCost;
                action.table.totalItem= action.table.totalItem- 1;
            return {
                ...state,
                ...newTable
            };
        
        case "ADD_ROW":
            const newRowInTable = action.table.item.push(action.menu)
            action.table.totalCost = action.table.totalCost + action.menu.Cost;
            action.table.totalItem= action.table.totalItem + 1;
            return {
                ...state,
                ...newRowInTable
            };

        case "ADD_ITEM":
            const items = [action.item, ...state.item]
            return {
                ...state,
                item: items,
            };
        case "ADD_REVIEW":
            const review = [action.review,...state.reviews]
            return{
                ...state,
                reviews:review
            };
        
        case "GET_STATE":
            return {  
                ...action.data,
            }

        case "DELETE_ALL_ITEM":
            const newTable1 = action.table.item.splice(0,action.table.item.length)
            action.table.totalCost = 0;
            action.table.totalItem = 0;
            return {
                ...state,
                ...newTable1
            }
            
        default:
            return state;    
    }
};

 export default itemReducer;