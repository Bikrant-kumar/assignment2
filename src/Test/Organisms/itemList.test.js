import React from 'react';
import { screen, render } from '@testing-library/react';
import ItemList from '../../Component/Organism/ItemList';
import ListReducer from '../../Store/Reducer/ListReducer';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import MockData from '../MockData';
import { createStore } from 'redux';

describe('ItemList',()=> {

    test('ItemList Render',() => {   
        const item=
        {
            "id": "item-27",
            "Name": "Chenna Poda",
            "Cost": 150,
            "Serving":1,
            "Photo":"https://encrypted-tbn0.gstatic.com/images?q=tbn%3AANd9GcRAv5Iol4J1UYRJ9sb5mqRV47D-v37ymZXe_A&usqp=CAU"
  };   
          render(
            <Provider store={createStore(ListReducer, MockData)}>
          <ItemList item={item} manager="true"/>
          </Provider>);
          expect(screen.getByText("MENU")).toBeInTheDocument();
       });
  
       test('test ItemList with snapshot',()=>{
         const item=
            {
                "id": "item-27",
                "Name": "Chenna Poda",
                "Cost": 150,
                "Serving":1,
                "Photo":"https://encrypted-tbn0.gstatic.com/images?q=tbn%3AANd9GcRAv5Iol4J1UYRJ9sb5mqRV47D-v37ymZXe_A&usqp=CAU"
      };  
        const component= renderer.create(
          <Provider store={createStore(ListReducer, MockData)}><ItemList item = {item} manager="true"/> </Provider>).toJSON();
        expect(component).toMatchSnapshot();
       });
  
  })
   