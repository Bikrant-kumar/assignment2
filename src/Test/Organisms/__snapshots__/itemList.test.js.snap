// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemList test ItemList with snapshot 1`] = `
Array [
  <div
    className="MuiBox-root MuiBox-root-3"
  >
    <div
      className="MuiBox-root MuiBox-root-4"
    >
      MENU
    </div>
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12"
    />
  </div>,
  " ",
]
`;
